sapi/fpm/fpm/zlog.c

```diff
@@ -678,6 +682,17 @@ ssize_t zlog_stream_prefix_ex(struct zlog_stream *stream, const char *function,
 		len = zlog_buf_prefix(
 				function, line, stream->flags,
 				stream->buf.data, stream->buf.size, stream->use_syslog);
+		if (!EXPECTED(len + 1 <= stream->buf.size)) {
```

Comment:
```suggestion
		if (UNEXPECTED(len + 1 > stream->buf.size)) {
```

Reply:
It's just a bit more readable for me but don't really mind it either way...

Reply:
Using UNEXPECTED may put the code on the cold text section and induce a cache miss, while using `!EXPECTED` just reorders the branches without moving the code to the cold text section. That's why I chose to use `!EXPECTED`.

> I guess it's a common mistake with snprintf

Yes it very much is

Reply:
Ah ok. Makes sense. Not exactly a hot path but why not optimize it... :)